### PR TITLE
Feat/delete shared post backend

### PR DESCRIPTION
**Pull request by jeromes123**
OVERVIEW
Backend for deleting shared post

### CODE REVIEW

File: src/app/Services/PostShareService.php [hunk 1]
```diff
@@ -27,4 +27,12 @@ public function getAllPostShares(): Collection
     {
         return PostShare::all();
     }
+
+    /**
+     * Deletes a shared post
+     */
+    public function delete(int $postShareId): void
+    {
+        PostShare::destroy($postShareId);
```
Comment (mikco-hipe) [inline src/app/Services/PostShareService.php:36]: are you implementing soft delete here?

Comment (jeromes123) [reply src/app/Services/PostShareService.php:36]: no sir

Comment (mikco-hipe) [reply src/app/Services/PostShareService.php:36]: Can you implement it?

Comment (jeromes123) [reply src/app/Services/PostShareService.php:36]: yes sir, will do

File: src/database/migrations/2024_05_20_094820_add_deleted_at_to_post_shares_table.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('post_shares', function (Blueprint $table) {
+            //
```
Comment (mikco-hipe) [inline src/database/migrations/2024_05_20_094820_add_deleted_at_to_post_shares_table.php]: is this supposed to be empty?

File: src/database/migrations/2024_05_20_094820_add_deleted_at_to_post_shares_table.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::table('post_shares', function (Blueprint $table) {
+            //
```
Comment (mikco-hipe) [inline src/database/migrations/2024_05_20_094820_add_deleted_at_to_post_shares_table.php]: is this supposed to be empty?

**Review by mikco-hipe — COMMENTED** [review]
Please check!

**Review by mikco-hipe — APPROVED** [review]
LGTM 🙇

**Pull request merged**